### PR TITLE
Image index with reversed order display feature added, test added

### LINKED ISSUE #5: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [ ] New images that are added show up on the homepage.
- [ ] These images are persisted if the browser is closed or even if the
  server is restarted.
- [ ] Images are not displayed wider than 400px.
- [ ] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by HongquanYu**
Closes #5 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [x] I have replaced `X` above with the associated issue number.
- [x] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [x] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by HongquanYu** [comment]
[Image Index feature](https://trello.com/c/ifedNZ6L/370-image-index-feature)

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -47,6 +47,52 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
     assert_equal 'Image with id 100000 does not exist', flash[:notice]
   end
 
+  test 'index page should display right amount of images' do
+    assert_equal 0, Image.count
+    @img_params4 = { title: 'bulldog', link: 'https://nhl.bamcontent.com/images/photos/301406224/1024x576/cut.jpg' }
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: Is `@img_params4` used outside of this test? If not, does it need to be an instance variable?

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -47,6 +47,52 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    post images_path, params: { image: @img_params4 }
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: I usually suggest avoiding making multiple HTTP requests in a controller test, particularly if it is for test set up. For example, if there is a bug in the `create` action this test will fail for reasons unrelated to the `index` action.

Can the creation of an image for test setup be achieved without needing to issue a post request?

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -47,6 +47,52 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+
+    get images_path
+
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: Remember to check the response codes of requests! (e.g. `assert_response :ok`)

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -47,6 +47,52 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+      assert_equal 1, imgs.count
+      assert_equal '400px', imgs[0].attr('width').to_s
+    end
+
+    assert_equal 1, Image.count
+  end
+
+  test 'index page images should displayed in reversed order by created time' do
+    @img_params5 = { title: 'bulldog', link: 'https://nhl.bamcontent.com/images/photos/301406224/1024x576/cut.jpg' }
+    @img_params6 = { title: 'puppy', link: 'https://cdn.orvis.com/images/DBS_SibHusky.jpg' }
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: Similar comment above about instance variables

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -47,6 +47,52 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+
+    post images_path, params: { image: @img_params5 }
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: Similar comment above about creating images without another HTTP request

File: test/controllers/images_controller_test.rb [hunk 6]
```diff
@@ -47,6 +47,52 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+      assert_equal 2, imgs.count
+
+      assert_equal '400px', imgs[0][:width]
+      assert_equal '400px', imgs[1][:width]
+      assert_equal 'https://cdn.orvis.com/images/DBS_SibHusky.jpg', imgs[0][:src]
+      assert_equal 'https://nhl.bamcontent.com/images/photos/301406224/1024x576/cut.jpg', imgs[1][:src]
+    end
+  end
+
+  test 'images should be persisted into test database' do
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: I think the test has already been covered in another test that tests the `create` action.

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (HongquanYu) [reply test/controllers/images_controller_test.rb]: changed

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (HongquanYu) [reply test/controllers/images_controller_test.rb]: Make sense, changed to insert data using utility methods

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (HongquanYu) [reply test/controllers/images_controller_test.rb]: Done!

File: test/controllers/images_controller_test.rb (hunk 4 above)
Comment (HongquanYu) [reply test/controllers/images_controller_test.rb]: Changed

File: test/controllers/images_controller_test.rb (hunk 5 above)
Comment (HongquanYu) [reply test/controllers/images_controller_test.rb]: Changed

File: test/controllers/images_controller_test.rb (hunk 6 above)
Comment (HongquanYu) [reply test/controllers/images_controller_test.rb]: Deleted

**Pull request merged**